src/libsyntax/parse/parser/item.rs

```diff
@@ -160,16 +160,27 @@ impl<'a> Parser<'a> {
         if self.eat_keyword(kw::Const) {
             let const_span = self.prev_span;
             if self.check_keyword(kw::Fn)
+                || self.check_keyword(kw::Extern)
                 || (self.check_keyword(kw::Unsafe)
-                    && self.is_keyword_ahead(1, &[kw::Fn])) {
+                    && (self.is_keyword_ahead(1, &[kw::Fn])
+                        || self.is_keyword_ahead(1, &[kw::Extern]))) {
                 // CONST FUNCTION ITEM
+
                 let unsafety = self.parse_unsafety();
-                self.bump();
+                if self.check_keyword(kw::Extern) {
```

Comment:
Can you move this to after parsing the `let abi` and check for `if let Some(_) = abi` instead?
(This will also have the effect that the span of `extern "C"` is included which I think it should.)

Reply:
`abi` isn't an option - it's either `Abi::Rust` or something else.

Also, calling `parse_extern_abi` eats the keyword, so that won't work.

Reply:
Ah whoops; that's right. Should use `self.token.span` instead then.